concent_api/core/migrations/0003_add_and_update_deadline_and_size.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-06-19 09:37
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db import models
+from common.helpers import deserialize_message
+from common.helpers import parse_timestamp_to_utc_datetime
+
+
+def populate_subtask_size_field(apps, _schema_editor):
+    Subtask = apps.get_model('core', 'Subtask')
+    StoredMessage = apps.get_model('core', 'StoredMessage')
+    for subtask in Subtask.objects.filter(result_package_size__isnull=True):
+        stored_message = StoredMessage.objects.filter(
+            pk=subtask.report_computed_task_id
+        ).first()
+        if stored_message is not None:
+            subtask.result_package_size = deserialize_message(
+                stored_message.data
+            ).size
+            subtask.full_clean()
+            subtask.save()
+
+
+def populate_deadline_size_field(apps, _schema_editor):
+    Subtask = apps.get_model('core', 'Subtask')
+    StoredMessage = apps.get_model('core', 'StoredMessage')
+    for subtask in Subtask.objects.filter(computation_deadline__isnull=True):
+        stored_message = StoredMessage.objects.filter(
+            pk=subtask.task_to_compute_id
+        ).first()
```

Comment:
If you expect only one item to be returned you should either use `get()` or `assert` that there's only one result after the `filter()`. As it is now if you happen to get more than one result you silently ignore it and take the first one.

For me seeing a call to `first()` is an instant red flag. I don't think I've ever had to use `first()` myself. If you use it anywhere else, please make sure you're not making the first error.

Reply:
Done.